src/main/java/io/github/opencubicchunks/cubicchunks/core/server/VanillaNetworkHandler.java

```diff
@@ -273,36 +339,47 @@ private void switchPlayerOffset(PlayerCubeMap cubeMap, EntityPlayerMP player, in
             } else {
                 lastSendCubes.add(cubeWatcher.getCube());
             }
+
+            clientLoadedChunks.add(new ChunkPos(cubeWatcher.getX() + offset.getX(), cubeWatcher.getZ() + offset.getZ()));
```

Comment:
You still need to take into account ColumnWatcher. CC will send empty columns with just biome data. Otherwise you are leaking client memory when switching offset before all cubes in the view radius have been generated.

Reply:
Good point, forgot about that.